api/v1alpha1/ext_auth_types.go

```diff
@@ -0,0 +1,108 @@
+)
+
+// +kubebuilder:validation:XValidation:message="http must be specified if type is HTTP",rule="self.type == 'HTTP' ? has(self.http) : true"
+// +kubebuilder:validation:XValidation:message="grpc must be specified if type is GRPC",rule="self.type == 'GRPC' ? has(self.grpc) : true"
+// +kubebuilder:validation:XValidation:message="only one of grpc or http can be specified",rule="!(has(self.grpc) && has(self.http))"
+//
+// ExtAuth defines the configuration for External Authorization.
+type ExtAuth struct {
+	// Type decides the type of External Authorization.
+	// Valid ExtAuthServiceType values are "GRPC" or "HTTP".
+	// +kubebuilder:validation:Enum=GRPC;HTTP
+	// +unionDiscriminator
+	Type ExtAuthServiceType `json:"type"`
+
+	// GRPC defines the gRPC External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	GRPC *GRPCExtAuthService `json:"grpc,omitempty"`
+
+	// HTTP defines the HTTP External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	HTTP *HTTPExtAuthService `json:"http,omitempty"`
+
+	// HeadersToExtAuth defines the client request headers that will be
+	// included in the request to the external authorization service.
+	// Note: If not specified, the default behavior of different external
+	// authorization services is different. All headers will be included in the
+	// check request to a gRPC authorization server, whereas no headers will be
+	// included in the check request to an HTTP authorization server.
+	// +optional
+	HeadersToExtAuth []string `json:"headersToExtAuth,omitempty"`
+}
+
+// GRPCExtAuthService defines the gRPC External Authorization service
+type GRPCExtAuthService struct {
+	// Host is the hostname of the gRPC External Authorization service
+	Host gwapiv1a2.Hostname `json:"host"`
+
+	// Port is the network port of the gRPC External Authorization service
+	Port gwapiv1a2.PortNumber `json:"port"`
+
+	// TLS defines the TLS configuration for the gRPC External Authorization service.
+	// Note: If not specified, the proxy will talk to the gRPC External
+	// Authorization service in plaintext.
+	// +optional
+	TLS *TLSConfig `json:"tls,omitempty"`
+}
+
+// HTTPExtAuthService defines the HTTP External Authorization service
+type HTTPExtAuthService struct {
+	// URL is the URL of the HTTP External Authorization service.
+	// The URL must be a fully qualified URL with a scheme, hostname,
+	// and optional port and path. Parameters are not allowed.
+	// The URL must use either the http or https scheme.
+	// If port is not specified, 80 for http and 443 for https are assumed.
+	// If path is specified, the authorization request will be sent to that path,
+	// or else the authorization request will be sent to the root path.
+	// +kubebuilder:validation:Pattern=`^(https?:)(\/\/[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)((:\d+)?)(\/[a-zA-Z0-9-._~%]+\/?)*$`
+	URL string `json:"url" yaml:"url"`
```

Comment:
This feels odd. Unless Envoy's extauth protocol has dramatically changed, the HTTP request made to an HTTP extauth implementation mirrors the original URL (optionally with an added prefix), so I think what we need here is the host & port rather than a full URL.